tests/irdl/test_attr_constraint.py

```diff
@@ -60,35 +61,37 @@ class AttrA(Base):
 class AttrB(Base):
     name = "test.attr_b"
 
-    param: ParameterDef[AttrA]
-
 
 @irdl_attr_definition
 class AttrC(Base):
     name = "test.attr_c"
 
+    param: ParameterDef[AttrA | AttrB]
+
 
```

Comment:
I renamed things to allow a test below but now realise that this has probably made the diff horiffic. Not sure how to fix really.

Reply:
If it's not too tedious, I think that it would be good to have separate tests for the case where the param is a union and not, so first rename the existing attrs (A(), B(), C(A)), then add D(A|B). WDYT? Or avoid renaming, and just add D(A|C)